src/torrent.cpp

```diff
@@ -4900,7 +4900,7 @@ bool is_downloading_state(int const st)
 		bool const was_finished = is_finished();
 		for (auto const& p : pieces)
 		{
-			static_assert(std::is_unsigned<decltype(p.second)>::value
+			static_assert(std::is_unsigned<decltype(p.second)::underlying_type>::value
```

Comment:
I would have thought you need to say `typename` here (to indicate `underlying_type` is a type, and not a value)